### PR TITLE
feat: Introduce AI dialog model "Kimi"

### PR DESCRIPTION
**Pull request by Fleurxxx**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced token management with dynamic retrieval and validation.
	- Introduced a new property `currentModel` in the token dialog for improved interactivity.
	- Added support for a new AI model, Kimi, with API key management.

- **Bug Fixes**
	- Updated validation messages for clearer user guidance regarding token input.
	- Improved handling of token status updates to enhance user experience.

### CODE REVIEW

File: packages/plugins/robot/src/Main.vue [hunk 1]
```diff
@@ -154,7 +159,8 @@ export default {
           : JSON.stringify({
               foundationModel: {
                 manufacturer: selectedModel.value.manufacturer,
-                model: selectedModel.value.value
+                model: selectedModel.value.value,
```
Comment (CatsAndMice) [inline packages/plugins/robot/src/Main.vue]: [3.3 组件设计](https://cn.vuejs.org/api/reactivity-utilities.html#unref) 使用这个方法后可以不需要.value。const unrefSelectedModel = unref(selectedModel)

File: packages/plugins/robot/src/TokenDialog.vue [hunk 2]
```diff
@@ -15,34 +15,37 @@ export default {
...
-      if (value.length > 100 || !accessTokenReg.test(value)) {
+    const tokenReg = /^[A-Za-z0-9\-.]+$/
+    const tokenValidate = (rule, value, callback) => {
+      if (value.length > 100 || !tokenReg.test(value)) {
         callback(new Error('参数错误，请输入小于100位的英文数字字符串'))
       } else {
         callback()
       }
     }
     const rules = ref({
```
Comment (CatsAndMice) [inline packages/plugins/robot/src/TokenDialog.vue]: 这个是校验表单规则，没必要ref设置成响应式数据吧？

File: packages/plugins/robot/src/js/robotSetting.js [hunk 3]
```diff
@@ -17,8 +17,9 @@ import { useHttp } from '@opentiny/tiny-engine-http'
 const { getBlockList } = useBlock()
 
 export const AIModelOptions = [
-  { label: 'ChatGPT：gpt-3.5-turbo', value: 'gpt-3.5-turbo', manufacturer: 'openai' },
-  { label: '文心一言：ERNIE-Bot-turbo', value: 'ERNIE-Bot-turbo', manufacturer: 'baiduai' }
+  { label: 'ChatGPT：gpt-3.5-turbo', value: 'gpt-3.5-turbo', manufacturer: 'openai', localKey: '' },
```
Comment (CatsAndMice) [inline packages/plugins/robot/src/js/robotSetting.js]: ChatGPT、文心一言现在填写一个token后能用吗？ 仅保存能用的

Comment (Fleurxxx) [reply packages/plugins/robot/src/js/robotSetting.js]: 文心一言可以用，chatGPT不可以所以暂时设为空了，是要在选择gpt时不弹出token填写的弹窗吗

Comment (CatsAndMice) [reply packages/plugins/robot/src/js/robotSetting.js]: > 文心一言可以用，chatGPT不可以所以暂时设为空了，是要在选择gpt时不弹出token填写的弹窗吗

用不了就不给用户看到这个选项了

**Review by CatsAndMice — APPROVED** [review]

File: packages/plugins/robot/src/js/robotSetting.js (hunk 3 above)
Comment (CatsAndMice) [reply packages/plugins/robot/src/js/robotSetting.js]: 别让用户看到不能用的Ai模型

**Comment by Fleurxxx** [comment]
@hexqi 可以帮我code review一下这个pr吗

File: packages/plugins/robot/src/Main.vue [hunk 4]
```diff
@@ -137,6 +142,7 @@ export default {
     const inputContent = ref('')
     const inProcesing = ref(false)
     const selectedModel = ref(AIModelOptions[0])
+    let unrefSelectedModel = unref(AIModelOptions[0])
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue]: 为什么要加一个unrefSelectedModel变量呢，并且这里AIModelOptions原本就不是响应式变量

File: packages/plugins/robot/src/js/robotSetting.js [hunk 5]
```diff
@@ -17,8 +17,10 @@ import { useHttp } from '@opentiny/tiny-engine-http'
 const { getBlockList } = useBlock()
 
 export const AIModelOptions = [
-  { label: 'ChatGPT：gpt-3.5-turbo', value: 'gpt-3.5-turbo', manufacturer: 'openai' },
-  { label: '文心一言：ERNIE-Bot-turbo', value: 'ERNIE-Bot-turbo', manufacturer: 'baiduai' }
+  // 暂时不能使用，预留模型信息
+  // { label: 'ChatGPT：gpt-3.5-turbo', value: 'gpt-3.5-turbo', manufacturer: 'openai', localKey: '' },
+  // { label: '文心一言：ERNIE-Bot-turbo', value: 'ERNIE-Bot-turbo', manufacturer: 'baiduai', localKey: 'ACCESS_TOKEN' },
```
Comment (hexqi) [inline packages/plugins/robot/src/js/robotSetting.js]: 文心一言这里为什么要屏蔽呢

File: packages/plugins/robot/src/Main.vue [hunk 6]
```diff
@@ -383,9 +394,21 @@ export default {
         })
       }
     }
+    const currentModel = ref(selectedModel.value)
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue]: selectedModel已经是ref变量了，这个currentModel作用是什么呢？

File: packages/plugins/robot/src/js/robotSetting.js [hunk 7]
```diff
@@ -17,8 +17,10 @@ import { useHttp } from '@opentiny/tiny-engine-http'
...
+  { label: 'Kimi：moonshot-v1-8k', value: 'moonshot-v1-8k', manufacturer: 'kimi', localKey: 'MOONSHOT_API_KEY' }
```
Comment (hexqi) [inline packages/plugins/robot/src/js/robotSetting.js]: localKey这个字段名感觉不是太恰当不能很好反应该字段作用，可以考虑换个名字

File: packages/plugins/robot/src/js/robotSetting.js (hunk 5 above)
Comment (Fleurxxx) [reply packages/plugins/robot/src/js/robotSetting.js]: 文心一言因为它上传图片是另一个token，所以暂时不可以实现图片上传，仅支持对话功能所以屏蔽掉，先用kimi把所有需求做完

**Review by hexqi — APPROVED** [review]

**Pull request merged**